src/plugins/intel_gna/src/gna_plugin.cpp

```diff
@@ -102,6 +102,9 @@
 
 #include <gna2-model-api.h>
 #include <gna2-common-api.h>
+#ifdef HAVE_AVX2
```

Comment:
Isn't something like following needed (in CMakeLists.txt)? otherwise HAVE_AVX2 won't be enabled at compile time within gna_plugin.cpp (?):
```
cross_compiled_file(${TARGET_NAME}
        ARCH AVX2 ANY
                    src/gna_plugin.cpp
```

Reply:
> Isn't something like following needed (in CMakeLists.txt)? otherwise HAVE_AVX2 won't be enabled at compile time within gna_plugin.cpp (?):
> 
> ```
> cross_compiled_file(${TARGET_NAME}
>         ARCH AVX2 ANY
>                     src/gna_plugin.cpp
> ```

I don't think so. It is relates only one particular file.

Reply:
> check how it's implemented in https://github.com/openvinotoolkit/openvino/tree/master/src/common/preprocessing in both cmake and C++ code

I've investigated this part as well (two compilation units handled with this way: ie_preprocess_data
.cpp.o, ie_preprocess_gapi.cpp.o), the purpose seemed to me to aim at another target then a *plugin*, but yes, @ilya-lavrenov thank you, I'll take it as a better way to go.
Anyway, common baseline is that this place requires a bit more attention.

Reply:
+ my feeling about "cross_compiled_file(${TARGET_NAME}" way, is that it ?tries? to handle architectures do support AVX2 and those that do not. I mean, during linking process both AVX2 and ANY are tried to be linked